cmd/odf/root/root.go

```diff
@@ -92,6 +110,16 @@ func getClientsets(ctx context.Context) *k8sutil.Clientsets {
 	return clientsets
 }
 
+func getControllerRuntimeClient() ctrl.Client {
+	client, err := ctrl.New(ClientSets.KubeConfig, ctrl.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		logging.Fatal(err)
+	}
+	return client
+}
```

Comment:
```suggestion
	client, err := ctrl.New(ClientSets.KubeConfig, ctrl.Options{
		Scheme: scheme,
	})
	if err != nil {
		logging.Fatal(err)
	}
	return client
}
```

@umangachapagain can we move this implementation inside `getClientsets` similar to other clients?

Reply:
controller-runtime clients are implemented differently from other ClientSets. We could even replace all the clients in the ClientSets with this single client. That is why I chose to keep it separate (it is not a ClientSet type).

I also looked at replacing other clientsets with this client to keep the dependencies minimal and easy to manage. But, the refactor was needed in the rook-ceph CLI as well (due to hard dependency between the projects). Given controller-runtime client is recommended over ClientSets, should we do a refactor in that direction? @travisn @subhamkrai

Reply:
> controller-runtime clients are implemented differently from other ClientSets. We could even replace all the clients in the ClientSets with this single client. That is why I chose to keep it separate (it is not a ClientSet type).
> 
> I also looked at replacing other clientsets with this client to keep the dependencies minimal and easy to manage. But, the refactor was needed in the rook-ceph CLI as well (due to hard dependency between the projects). Given controller-runtime client is recommended over ClientSets, should we do a refactor in that direction? @travisn @subhamkrai

@umangachapagain yes, sounds good to do the refactoring.

Reply:
Will you open a follow-up PR for this?

Reply:
Sure. Might take a while, but will do.

Reply:
sure, Thanks